model_api/cpp/models/src/segmentation_model.cpp

```diff
@@ -54,6 +68,34 @@ cv::Mat create_hard_prediction_from_soft_prediction(const cv::Mat& soft_predicti
     }
     return hard_prediction;
 }
+
+std::vector<Annotation> create_annotation_from_segmentation_map(const cv::Mat& hard_prediction, const cv::Mat& soft_prediction, std::vector<std::string> label_map) {
+    std::vector<Annotation> annotations = {};
+    cv::Mat label_index_map;
+    cv::Mat current_label_soft_prediction;
+    size_t labelID = 0;
+    for (int index = 1; index < soft_prediction.channels(); index++) {
+        cv::extractChannel(soft_prediction, current_label_soft_prediction, index);
+        cv::inRange(hard_prediction, cv::Scalar(index, index, index), cv::Scalar(index, index, index), label_index_map);
+        std::vector<std::vector<cv::Point>> contours;
+        cv::findContours(label_index_map, contours, cv::RETR_EXTERNAL, cv::CHAIN_APPROX_NONE);
+        for (auto &contour: contours) {
+            cv::Mat mask = cv::Mat::zeros(hard_prediction.rows, hard_prediction.cols, hard_prediction.type());
+            std::vector<std::vector<cv::Point>> localContours = { contour };
+            cv::drawContours(mask, localContours, -1, 255, -1);
+            float probability = cv::mean(current_label_soft_prediction, mask)[0];
+            annotations.push_back( {
+                    labelID,
+                    label_map[index],
```

Comment:
You must handle the case when `label_map.size() >= index`. See https://github.com/openvinotoolkit/model_api/blob/e86a632d2a43bfa14d326c431023a1979f693b56/model_api/cpp/models/include/models/detection_model.h#L46

Reply:
Yes, this also caused the error in the test. I've moved the getLabelName to ImageModel and the get_annotations to segmentation_model class so that I can reuse it.